smmips/smmips.py

```diff
@@ -57,16 +96,143 @@ def assign_smmips(outdir, fastq1, fastq2, reference, bwa, prefix, remove, panel,
     
     # align fastqs
     prefix = os.path.basename(prefix)
-    sortedbam = align_fastqs(fastq1, fastq2, reference, outdir, bwa, prefix, remove)
+    
+    # open files for writing
+    
+    # create AlignmentFile object to read input bam
+    infile = pysam.AlignmentFile(sortedbam, 'rb')
+    # create AlignmentFile objects for writing reads
+    if chromosome == 'all':
+        # create a new file, use header from bamfile
+        assigned_filename = remove_bam_extension(sortedbam) + '.assigned_reads.bam'
+        # open bams for writing unassigned reads. use header from bamfile
+        unassigned_filename = remove_bam_extension(sortedbam) + '.unassigned_reads.bam'
+        # open bam for writing assigned but empty reads
+        empty_filename = remove_bam_extension(sortedbam) + '.empty_reads.bam'
+    else:
```

Comment:
here it opens files for each chromosome
1. where does it get the list of chromosomes to process?  from the aligned sequence?  if this is the case, then it will attempt to process some chromosomes for which there are no smmips designed.  i think it should be running for each of the chromosomes in the smmip panel, and any reads aligning to other chromosomes should simply be discarded (or saved in a discard file)

2. how will this run in a workflow.  Can we give the tool a chromosome to process, and use the scatter/gather approach in wdl?, relying on the combine aspect of this tool for the gather part?

Reply:
The list of chromosomes comes from the panel. The workflow uses the scatter approach with an array of chromosomes to parallelize QC.